FosUser/EventListener.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the DunglasApiBundle package.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Dunglas\ApiBundle\FosUser;
+
+use FOS\UserBundle\Model\UserManagerInterface;
+use FOS\UserBundle\Model\UserInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\GetResponseForControllerResultEvent;
+
+/**
+ * Bridges between FOSUserBundle and DunglasApiBundle.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ * @author Théo FIDRY <theo.fidry@gmail.com>
+ */
+class EventListener
+{
+    /**
+     * @var UserManagerInterface
+     */
+    private $userManager;
+
+    public function __construct(UserManagerInterface $userManager)
+    {
+        $this->userManager = $userManager;
+    }
+
+    /**
+     * Persists, updates or delete data return by the controller if applicable.
+     *
+     * @param GetResponseForControllerResultEvent $event
+     *
+     * @return mixed
+     */
+    public function onKernelView(GetResponseForControllerResultEvent $event)
+    {
+        $controllerResult = $event->getControllerResult();
+        if (!$controllerResult instanceof UserInterface) {
+            return $controllerResult;
+        }
+
+        switch ($event->getRequest()->getMethod()) {
+
+            case Request::METHOD_POST:
+            case Request::METHOD_PUT:
+                $this->userManager->updateUser($controllerResult, false);
```

Comment:
I'm not really fond of that this listener cares of the persistence layer. Expect minor performance, there's no gain IMO as if the user needs an other listener, it can just add it before this one.